src/main/scala/viper/server/vsi/Cache.scala

```diff
@@ -46,10 +48,11 @@ abstract class Cache {
     * The inner map is referred to as the fileCache. As the name indicates, it stores, for each
     * file, a number of hashes and corresponding cache entries.
     */
-  protected val _cache: MutableMap[String, FileCash] = MutableMap()
-  type FileCash = MutableMap[String, List[CacheEntry]]
+  protected var _cache: Cache = Map()
+  type Cache = Map[String, FileCache]
 
-  protected val _program_cache: MutableMap[Ast, MutableMap[CacheableMember, List[Member]]] = MutableMap()
+  type FileCache = Map[String, List[CacheEntry]]
+  protected var _program_cache: Map[String, Map[String, String]] = Map()
```

Comment:
could you please define additional types such that this becomes more readable? e.g. `Map[FileKey, DependencyMap]` where `type DependencyMap = Map[MemberHash, DependentMemberHash]`